### PR TITLE
style(#2689): Polish code and extract commonalities

### PR DESCRIPTION
**Pull request by RovingSea**
### Describe what this PR does / why we need it
#2689 

### Does this pull request fix one issue?
NONE

### Describe how you did it
- Increase the readability of collect uris
- Extract the same operation of extending ServiceMeta List
- Use generics to indicate types

### Describe how to verify it
This is a pre startup operation, so you can verify when you start

### CODE REVIEW

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java [hunk 1]
```diff
@@ -46,68 +47,82 @@ public final class URIRegister {
...
+
+	/**
+	 * Expand {@link ServiceMeta} list while determining whether it is a wild char.
+	 * @param serviceMetaList extended list
+	 * @param hasWildChar keyword to be determined
+	 * @param urlPatterns looped list
+	 * @param resourceActiveType attribute of {@link ServiceMeta}
+	 * @return is that a wild char
+	 */
+	private static boolean expandAndDetermine(List<ServiceMeta> serviceMetaList, boolean hasWildChar,
```
Comment (steverao) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java]: I think called `collectServiceMeta` is better. this is the purpose of the logic.

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java [hunk 2]
```diff
@@ -46,68 +47,82 @@ public final class URIRegister {
...
+		serviceMetaObject.setMd5OfList(md5);
+	}
+
+	/**
+	 * Expand {@link ServiceMeta} list while determining whether it is a wild char.
+	 * @param serviceMetaList extended list
+	 * @param hasWildChar keyword to be determined
+	 * @param urlPatterns looped list
+	 * @param resourceActiveType attribute of {@link ServiceMeta}
+	 * @return is that a wild char
```
Comment (steverao) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java]: @return is new wildChar, please modified relevant parameters description depend on new method name.

Comment (RovingSea) [reply spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java]: hi @steverao, I have improved the document.

**Comment by steverao** [comment]
LGTM.

**Comment by steverao** [comment]
Don't create any PR to branch of 2.2.x-appactive and I merge your PR to 2.2.x at same time, all codes are in main branch of 2.2.x and we will delete 2.2.x-appactive later.

**Comment by RovingSea** [comment]
> Don't create any PR to branch of 2.2.x-appactive and I merge your PR to 2.2.x at same time, all codes are in main branch of 2.2.x and we will delete 2.2.x-appactive later.

Ok, I see.

**Pull request merged**